File: python/stopwatch.py
import time
import sys

def convert_secs(sec):
    mins = sec // 60
    sec = sec % 60
    hours = mins // 60
    mins = mins % 60
    return (int(hours), int(mins), int(sec))

def clear_line():
    sys.stdout.write("\033[F")

if __name__ == '__main__':
    time_lapsed = 0
    while True:
        (hours, mins, sec) = convert_secs(time_lapsed)
        print(f"Time lapsed = {hours:02}:{mins:02}:{sec:02}")
        clear_line()
        time.sleep(1)
        time_lapsed = time_lapsed + 1

File: javascript/stopwatch.js
class Stopwatch {
  convertSeconds(secs) {
    let mins = Math.floor(secs / 60);
    secs %= 60;
    const hours = Math.floor(mins / 60);
    mins %= 60;
    return [hours, mins, secs];
  }
  
  startStopwatch() {
    let timeLapsed = 0;
    setInterval(() => {
      this.clearLine();
      const time = this.convertSeconds(timeLapsed);
      this.printTime(time);
      timeLapsed+=180;
    }, 1000)
  }

  printTime(time) {
    const twoDigit = (num) => (num < 10 ? "0" : "") + num;
    process.stdout.write(
      `${twoDigit(time[0])}:${twoDigit(time[1])}:${twoDigit(time[2])}`
    );
  }

  clearLine() {
    process.stdout.clearLine();
    process.stdout.cursorTo(0);
  }
}

const stopwatch = new Stopwatch();
stopwatch.startStopwatch();